core/utils/clock.go

```diff
@@ -0,0 +1,29 @@
+package utils
+
+import "time"
+
+type Clock interface {
+	Now() time.Time
+}
+
+type realClock struct{}
+
+func NewRealClock() Clock {
+	return &realClock{}
+}
+
+func (realClock) Now() time.Time {
+	return time.Now()
+}
+
+type fixedClock struct {
+	now time.Time
+}
+
+func NewFixedClock(now time.Time) Clock {
```

Comment:
Rather than fixed, I'd rather create a clock that you can set to whatever you want via a Set() method. But that can be added later if there's a need in tests.

Reply:
The clock is called fixed because it has fixed (immutable) time. You can set to whatever you want upon creation. If you are looking for a "mutable" fixed clock, that would be another type of clock. And yes, this type can be added later if needed.